### DISCUSSION TITLE
Optimization of MCS calculation

### DISCUSSION THREAD

**Discussion by mariamartinezs** [rdkit/rdkit#5745] [Q&A]
Hi!

I am trying to run a code in which I need to perform a pairwise similarity of a set of molecules of a database and calculate their MCS if they meet a condition. At first, I was doing this using itertools from python, but I have thought that maybe using the rdkit cartridge it is faster. 

I have succeeded on calculating the pairwise similarity with a cross join, but I would like to compute also their MCS and I cannot make the function fmcs work. 

```
select foo.mol_inchikey, boo.mol_inchikey, foo.mol_ecfp4_sfp, boo.mol_ecfp4_sfp, dice_sml(foo.mol_ecfp4_sfp, boo.mol_ecfp4_sfp) as sml, fmcs_smiles(foo.mol_mol::text, boo.mol_mol::text) as mcs
from (select distinct(m.mol_inchikey), m.mol_ecfp4_sfp, m.mol_mol 
		from cipsi_prototype.molecule m 
		inner join cipsi_prototype.pat2mol pm on pm.mol_inchikey = m.mol_inchikey) as foo
cross join (select distinct(m.mol_inchikey), m.mol_ecfp4_sfp, m.mol_mol 
		from cipsi_prototype.molecule m 
		inner join cipsi_prototype.pat2mol pm on pm.mol_inchikey = m.mol_inchikey) as boo
```

If this is not possible, is there a way to improve the performance of the MCS calculation?

```
rdFMCS.FindMCS([molecules[mol1]['mol'],
                                molecules[mol2]['mol']],
                                ringMatchesRingOnly=True, completeRingsOnly=True, timeout=1).smartsString
```

Thank you

**Comment by kienerj** [comment]
Can't comment if this is possible in the cartridge. In python I would use multiprocessing (or comparable tool) to spread the workload over all available CPU cores.
